### PR TITLE
Bugfix/1189 xml representation mapping

### LINKED ISSUE #1189: Incomplete mapping of ElementDefiniton.representation in StructureDefinitionSummaryProvider
The "Representation" attribute of StructureDefinitionSummaryProvider does not provide a complete mapping of all options available for ElementDefiniton.representation. This information is lost in cases where a logical model decides to use a different values besides "xmlAttr". For example the CCDA on FHIR logical models make use of "typeAttr", which is now mapped by default to "xmlAttr".

### PR DESCRIPTION
**Pull request by alexzautke**
Adjust StructureDefinitionSummaryProvider to return a complete mapping of ElementDefinition.representation.

Fixes #1189

### CODE REVIEW

File: src/Hl7.Fhir.Specification.Tests/StructureDefinitionSerializationInfoTests.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@ public class StructureDefinitionSerializationInfoTests
         [ClassInitialize]
         public static void SetupSource(TestContext t)
         {
-            source = ZipSource.CreateValidationSource();
+            source = new MultiResolver(ZipSource.CreateValidationSource(), new DirectorySource("TestData", includeSubdirectories: true));
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Specification.Tests/StructureDefinitionSerializationInfoTests.cs]: To construct DirectorySource like this is obsolete: use 
`new DirectorySource("TestData", new DirectorySourceSettings(includeSubdirectories: true))`

File: src/Hl7.Fhir.Specification.Tests/SerializationInfoTestsHelpers.cs [hunk 2]
```diff
@@ -175,5 +175,21 @@ int fail(string message)
...
+        public static void TestXmlRepresetation(IStructureDefinitionSummaryProvider provider)
+        {
+            var testElements = new XmlRepresentation[] { XmlRepresentation.XmlAttr, XmlRepresentation.TypeAttr, XmlRepresentation.XHtml }; // Elements defined by TestXmlRepresentation (Logical Model)
+            var summary = provider.Provide("http://hl7.org/fhir/StructureDefinition/TestXmlRepresentation");
+            var summaryElements = summary.GetElements();
+
+            foreach(var element in testElements)
+            {
+                var representation = summaryElements.Where(e => e.ElementName == element.ToString()).FirstOrDefault().Representation;
+                if (!representation.ToString().Equals(element.ToString()))
```
Comment (marcovisserFurore) [inline src/Hl7.Fhir.Specification.Tests/SerializationInfoTestsHelpers.cs]: Why not like this:
`if (!representation.Equals(element))`

File: src/Hl7.Fhir.Specification.Tests/StructureDefinitionSerializationInfoTests.cs (hunk 1 above)
Comment (alexzautke) [reply src/Hl7.Fhir.Specification.Tests/StructureDefinitionSerializationInfoTests.cs]: Changed as requested

File: src/Hl7.Fhir.Specification.Tests/SerializationInfoTestsHelpers.cs (hunk 2 above)
Comment (alexzautke) [reply src/Hl7.Fhir.Specification.Tests/SerializationInfoTestsHelpers.cs]: Of course... Changed as requested

**Pull request merged**